app/pods/components/cart-dialog/component.js

```diff
@@ -0,0 +1,28 @@
+import Ember from 'ember';
+import { task } from 'ember-concurrency';
+import { service } from 'ember-decorators/service';
+import { action } from 'ember-decorators/object'
+import env from 'codingblocks-online/config/environment';
+
+export default class extends Ember.Component {
+  @service api
+
+  dukaanUrl = env.dukaanUrl
+
+  addCartTask = task(function *(){
+    yield this.get('api').request('/runs/clear_cart');
+    const runId = this.get('run.id');
+    yield this.get('api').request(`/runs/${runId}/buy`);
+    window.location.href = this.get('dukaanUrl')
+  })
+
+  didReceiveAttrs() {
+    this.set('run', this.get('run'))
+    this.set('dukaanCart', this.get('dukaanCart'))
+  }
+
+  @action
+  onClose() {
```

Comment:
Might as well just directly bind it in template. No need to define a wrapper action here.
Check this twiddle here: https://ember-twiddle.com/f82322578e9beedcb148986938b7ab78?openFiles=controllers.application.js%2Ctemplates.components.modal-comp.hbs